app/models/concerns/statable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Statable
+  extend ActiveSupport::Concern
+  def notes_for_batch_process(batch_process_id)
+    return if self.class == Integer
+    note_records = Notification.where(["params->>'batch_process_id' = :id and
+    params->>'parent_object_id' = :oid", { id: batch_process_id.to_s, oid:
+    oid.to_s }])
+    notes = {}
```

Comment:
This could replace lines 10-12

```suggestion
    note_records.inject({}) { |i, n| i[n.params[:status]] = n.created_at; i }
```